Documentation improvements, reorganize examples as folders. (#49)
google/jaxopt: docs/fixed_point.rst

    jaxopt.AndersonAcceleration

Anderson acceleration is an iterative method that aims to compute the next
iterate :math:`x_{n}` as a linear combination of the :math:`m` last iterates

> :math:`[x_{n-m},x_{n-m+1},\ldots x_{n-1}]`. The optimal coefficients of the

Reviewer: I would remove optimal from here since we're not really explaining in which way we're optimal. I would say "The coefficients of the ..."

linear combination are computed 'on the fly' at each iteration. As a result,
not only the convergence is faster but the convergence conditions are weakened,

> allowing to tackle problems PicardIterations could not.  See `this paper

Reviewer: ```suggestion
allowing to tackle problems PicardIterations could not.  See `(Pollock and Rebholz, 2020)
```

<https://arxiv.org/abs/1909.04638>`_ for more details.

The size of the history :math:`m` (denoted ``history_size`` below) plays a

> crucial role in method performance.  An higher :math:`m` could speed up the

Reviewer: ```suggestion
crucial role in the method's performance.  An higher :math:`m` could speed up the
```

  print(jax.grad(fixed_point, argnums=1)(x_init, theta))  # only gradient
  print(jax.value_and_grad(fixed_point, argnums=1)(x_init, theta))  # both value and gradient

.. topic:: Examples

  * :ref:`sphx_glr_auto_examples_fixed_point_plot_anderson_accelerate_gd.py`
  * :ref:`sphx_glr_auto_examples_fixed_point_plot_picard_ode.py`

Equivalence with root finding
-----------------------------

Note that if :math:`x` is a fixed point of :math:`T` then :math:`x` is a root

> of :math:`F(x, \theta) = T(x, \theta) - x`.  Reciproqually, if :math:`x` is the

Reviewer: ```suggestion
of :math:`F(x, \theta) = T(x, \theta) - x`.  Reciprocally, if :math:`x` is the
```
